readme.md
# PROJECT 

This is running using a docker container, which is all set up. All you need to run for dev work is:
` docker-compose up -d `
 
** note: it's probably already running unless you did ` docker-compose down`

(to see you can type docker ps but whatevs)

then just navigate to localhost:8000 and do yer stuff

guides for theme:
/Users/merecoulter/repos/goldenboyphoto/kreativa-packed/Help Guides



kreativa-packed/Changelog.txt
---------------------------
Version 1.7
---------------------------

Proofing Archives and Taxonomy
Theme Options controls for Proofing archives

Added /kreativa/archive-mtheme_proofing.php
Modified /kreativa/framework/admin/tgm/tgm-init.php
Modified /kreativa/framework/options/options-data.php
Modified /kreativa/style.css
Added /kreativa/taxonomy-proofingsection.php
Modified /kreativa/template-parts/header-title.php

Plugins
'framework/plugins/imaginem-shortcodes-r11.zip

---------------------------
Version 1.6
---------------------------

Fix: To tabs for PHP 7
Minor adjustments to Pagebuilder

Files 'framework/admin/css/style.css'
Files 'framework/admin/tgm/tgm-init.php'
Files 'framework/functions/theme-functions.php'
Files 'framework/options/admin/css/admin-style.css'
Files 'js/jquery.tubular.1.0.js'
Files 'js/tilt.jquery.js'
Files 'style.css'

Plugins
'framework/plugins/imaginem-builder-r6.zip
'framework/plugins/imaginem-shortcodes-r11.zip

---------------------------
Version 1.5
---------------------------

Add Events choise for "Past Event"
Add Events pabebuilder block
Fix Thumbnails Gallery filter to display only the filtered lightboxes

Files 'framework/admin/js/postmetaboxes.js'
Files 'framework/admin/tgm/tgm-init.php'
Files 'framework/functions/theme-functions.php'
Files 'framework/metaboxes/events-metaboxes.php'
Files 'framework/metaboxes/portfolio-metaboxes.php'
Files 'framework/options/admin/js/init.js'
Files 'js/page-elements.js'
Files 'style.css'
Files 'template-parts/fullscreen/fullscreen-supersized.php'

Plugins
'framework/plugins/imaginem-builder-r6.zip
'framework/plugins/imaginem-shortcodes-r11.zip

---------------------------
Version 1.4
---------------------------

Fix: Blog small layout from pagebuilder
Fix: WooCommerce style corrections

Files 'css/responsive.css'
Files 'framework/admin/tgm/tgm-init.php'
Files 'style-dark.css'
Files 'style.css'

Plugins
'framework/plugins/imaginem-shortcodes-r11.zip

---------------------------
Version 1.3
---------------------------

Instagram grid fix for non square images
Lightbox title on/off via theme options

Files 'css/dynamic_css.php'
Files 'framework/admin/tgm/tgm-init.php'
Files 'framework/options/options-data.php'
Files 'style.css'
Files 'template-parts/header-title.php'

Plugins
'framework/plugins/imaginem-shortcodes-r11.zip

---------------------------
Version 1.2
---------------------------

WooCommerce variations fix
Thumbnail galleries minor fixes
Pagebuilder minor style fixes

Files 'framework/admin/tgm/tgm-init.php'
Files 'includes/background/fullscreen-slideshow.php'
Files 'style-dark.css'
Files 'style.css'
Files 'woocommerce/single-product/product-image.php'
Files 'woocommerce/single-product/product-thumbnails.php'

Plugins
'framework/plugins/imaginem-builder-r6.zip
'framework/plugins/imaginem-shortcodes-r11.zip

---------------------------
Version 1.1
---------------------------

Pagebuiler Upgrade
Portfolio permalink slug change choice
Photowall lightbox thumbnails fix

Files 'framework/admin/tgm/tgm-init.php'
Files 'style.css'
Files 'template-parts/fullscreen/fullscreen-photowall.php'

Plugins
'framework/plugins/imaginem-builder-r6.zip
'framework/plugins/imaginem-portfolio-post-type-r5.zip

---------------------------
Version 1.0.3
---------------------------
Minor transition changes in Horizontal fullscreen toggle
Fotorama thumbnails centered in Horizontal menu style

Files 'style.css'

---------------------------
Version 1.0.2
---------------------------

ADD: Horizontal Menu

Files 'css/dynamic_css.php'
Files 'css/responsive.css'
Files 'footer.php'
Files 'framework/functions/theme-functions.php'
Files 'framework/options/[all files and folders]
Files 'functions.php'
Files 'header.php'
Files 'js/common.js'
Files 'js/page-elements.js'
Files 'search.php'
Files 'style-dark.css'
Files 'style.css'
Files 'template-parts/fullscreen/fullscreen-supersized.php'
Files 'template-parts/header-navigation.php'
Files 'template-parts/header-title.php'

---------------------------
Version 1.0.1
---------------------------

Added One Click Demo Importer

Files 'framework/admin/tgm/tgm-init.php'
Files 'framework/options/admin/options-framework.php'
Files 'framework/options/options-data.php'
Files 'style.css'

Plugins
'framework/plugins/kreativa-demo-importer.zip

---------------------------
Version 1.0 initial release
---------------------------

wp-content/themes/kreativa/framework/admin/tgm/tgm-init.php
<?php
/**
 * This file represents an example of the code that themes would use to register
 * the required plugins.
 *
 * It is expected that theme authors would copy and paste this code into their
 * functions.php file, and amend to suit.
 *
 * @see http://tgmpluginactivation.com/configuration/ for detailed documentation.
 *
 * @package    TGM-Plugin-Activation
 * @subpackage Example
 * @version    2.6.1 for parent theme kreativa for publication on ThemeForest
 * @author     Thomas Griffin, Gary Jones, Juliette Reinders Folmer
 * @copyright  Copyright (c) 2011, Thomas Griffin
 * @license    http://opensource.org/licenses/gpl-2.0.php GPL v2 or later
 * @link       https://github.com/TGMPA/TGM-Plugin-Activation
 */

/**
 * Include the TGM_Plugin_Activation class.
 *
 * Depending on your implementation, you may want to change the include call:
 *
 * Parent Theme:
 * require_once get_template_directory() . '/path/to/class-tgm-plugin-activation.php';
 *
 * Child Theme:
 * require_once get_stylesheet_directory() . '/path/to/class-tgm-plugin-activation.php';
 *
 * Plugin:
 * require_once dirname( __FILE__ ) . '/path/to/class-tgm-plugin-activation.php';
 */
require_once get_template_directory() . '/framework/admin/tgm/class-tgm-plugin-activation.php';

add_action( 'tgmpa_register', 'kreativa_register_required_plugins' );

/**
 * Register the required plugins for this theme.
 *
 * In this example, we register five plugins:
 * - one included with the TGMPA library
 * - two from an external source, one from an arbitrary source, one from a GitHub repository
 * - two from the .org repo, where one demonstrates the use of the `is_callable` argument
 *
 * The variables passed to the `tgmpa()` function should be:
 * - an array of plugin arrays;
 * - optionally a configuration array.
 * If you are not changing anything in the configuration array, you can remove the array and remove the
 * variable from the function call: `tgmpa( $plugins );`.
 * In that case, the TGMPA default settings will be used.
 *
 * This function is hooked into `tgmpa_register`, which is fired on the WP `init` action on priority 10.
 */
function kreativa_register_required_plugins() {
	/*
	 * Array of plugin arrays. Required keys are name and slug.
	 * If the source is NOT from the .org repo, then source is also required.
	 */
	$plugins = array(

		array(
			'name'     				=> 'iMaginem Widget Set r9', // The plugin name
			'slug'     				=> 'imaginem-widgets-r9', // The plugin slug (typically the folder name)
			'source'   				=> get_template_directory() . '/framework/plugins/imaginem-widgets-r9.zip', // The plugin source
			'required' 				=> true, // If false, the plugin is only 'recommended' instead of required
			'version' 				=> '1.0', // E.g. 1.0.0. If set, the active plugin must be this version or higher, otherwise a notice is presented
			'force_activation' 		=> false, // If true, plugin is activated upon theme activation and cannot be deactivated until theme switch
			'force_deactivation' 	=> false, // If true, plugin is deactivated upon theme switch, useful for theme-specific plugins
			'external_url' 			=> '', // If set, overrides default API URL and points to an external URL
			'is_callable'        => '', // If set, this callable will be be checked for availability to determine if a plugin is active.
		),
		array(
			'name'     				=> 'iMaginem Fullscreen Post Type r4', // The plugin name
			'slug'     				=> 'imaginem-fullscreen-post-type-r4', // The plugin slug (typically the folder name)
			'source'   				=> get_template_directory() . '/framework/plugins/imaginem-fullscreen-post-type-r4.zip', // The plugin source
			'required' 				=> true, // If false, the plugin is only 'recommended' instead of required
			'version' 				=> '1.0', // E.g. 1.0.0. If set, the active plugin must be this version or higher, otherwise a notice is presented
			'force_activation' 		=> false, // If true, plugin is activated upon theme activation and cannot be deactivated until theme switch
			'force_deactivation' 	=> false, // If true, plugin is deactivated upon theme switch, useful for theme-specific plugins
			'external_url' 			=> '', // If set, overrides default API URL and points to an external URL
			'is_callable'        => '', // If set, this callable will be be checked for availability to determine if a plugin is active.
		),
		array(
			'name'     				=> 'iMaginem Portfolio Post Type r5', // The plugin name
			'slug'     				=> 'imaginem-portfolio-post-type-r5', // The plugin slug (typically the folder name)
			'source'   				=> get_template_directory() . '/framework/plugins/imaginem-portfolio-post-type-r5.zip', // The plugin source
			'required' 				=> true, // If false, the plugin is only 'recommended' instead of required
			'version' 				=> '1.1', // E.g. 1.0.0. If set, the active plugin must be this version or higher, otherwise a notice is presented
			'force_activation' 		=> false, // If true, plugin is activated upon theme activation and cannot be deactivated until theme switch
			'force_deactivation' 	=> false, // If true, plugin is deactivated upon theme switch, useful for theme-specific plugins
			'external_url' 			=> '', // If set, overrides default API URL and points to an external URL
			'is_callable'        => '', // If set, this callable will be be checked for availability to determine if a plugin is active.
		),
		array(
			'name'     				=> 'iMaginem Shortcodes r11', // The plugin name
			'slug'     				=> 'imaginem-shortcodes-r11', // The plugin slug (typically the folder name)
			'source'   				=> get_template_directory() . '/framework/plugins/imaginem-shortcodes-r11.zip', // The plugin source
			'required' 				=> true, // If false, the plugin is only 'recommended' instead of required
			'version' 				=> '1.6', // E.g. 1.0.0. If set, the active plugin must be this version or higher, otherwise a notice is presented
			'force_activation' 		=> false, // If true, plugin is activated upon theme activation and cannot be deactivated until theme switch
			'force_deactivation' 	=> false, // If true, plugin is deactivated upon theme switch, useful for theme-specific plugins
			'external_url' 			=> '', // If set, overrides default API URL and points to an external URL
			'is_callable'        => '', // If set, this callable will be be checked for availability to determine if a plugin is active.
		),
		array(
			'name'     				=> 'iMaginem Events r5', // The plugin name
			'slug'     				=> 'imaginem-events-r5', // The plugin slug (typically the folder name)
			'source'   				=> get_template_directory() . '/framework/plugins/imaginem-events-r5.zip', // The plugin source
			'required' 				=> true, // If false, the plugin is only 'recommended' instead of required
			'version' 				=> '1.0', // E.g. 1.0.0. If set, the active plugin must be this version or higher, otherwise a notice is presented
			'force_activation' 		=> false, // If true, plugin is activated upon theme activation and cannot be deactivated until theme switch
			'force_deactivation' 	=> false, // If true, plugin is deactivated upon theme switch, useful for theme-specific plugins
			'external_url' 			=> '', // If set, overrides default API URL and points to an external URL
			'is_callable'        => '', // If set, this callable will be be checked for availability to determine if a plugin is active.
		),
		array(
			'name'     				=> 'iMaginem Photo Proofing r4', // The plugin name
			'slug'     				=> 'imaginem-photo-proofing-r4', // The plugin slug (typically the folder name)
			'source'   				=> get_template_directory() . '/framework/plugins/imaginem-photo-proofing-r4.zip', // The plugin source
			'required' 				=> true, // If false, the plugin is only 'recommended' instead of required
			'version' 				=> '1.0', // E.g. 1.0.0. If set, the active plugin must be this version or higher, otherwise a notice is presented
			'force_activation' 		=> false, // If true, plugin is activated upon theme activation and cannot be deactivated until theme switch
			'force_deactivation' 	=> false, // If true, plugin is deactivated upon theme switch, useful for theme-specific plugins
			'external_url' 			=> '', // If set, overrides default API URL and points to an external URL
			'is_callable'        => '', // If set, this callable will be be checked for availability to determine if a plugin is active.
		),
		array(
			'name'     				=> 'iMaginem Pagebuilder r6', // The plugin name
			'slug'     				=> 'imaginem-builder-r6', // The plugin slug (typically the folder name)
			'source'   				=> get_template_directory() . '/framework/plugins/imaginem-builder-r6.zip', // The plugin source
			'required' 				=> true, // If false, the plugin is only 'recommended' instead of required
			'version' 				=> '1.4', // E.g. 1.0.0. If set, the active plugin must be this version or higher, otherwise a notice is presented
			'force_activation' 		=> false, // If true, plugin is activated upon theme activation and cannot be deactivated until theme switch
			'force_deactivation' 	=> false, // If true, plugin is deactivated upon theme switch, useful for theme-specific plugins
			'external_url' 			=> '', // If set, overrides default API URL and points to an external URL
			'is_callable'        => '', // If set, this callable will be be checked for availability to determine if a plugin is active.
		),
		array(
			'name'     				=> 'Kreativa Demo Importer', // The plugin name
			'slug'     				=> 'kreativa-demo-importer', // The plugin slug (typically the folder name)
			'source'   				=> get_template_directory() . '/framework/plugins/kreativa-demo-importer.zip', // The plugin source
			'required' 				=> true, // If false, the plugin is only 'recommended' instead of required
			'version' 				=> '1.0', // E.g. 1.0.0. If set, the active plugin must be this version or higher, otherwise a notice is presented
			'force_activation' 		=> false, // If true, plugin is activated upon theme activation and cannot be deactivated until theme switch
			'force_deactivation' 	=> true, // If true, plugin is deactivated upon theme switch, useful for theme-specific plugins
			'external_url' 			=> '', // If set, overrides default API URL and points to an external URL
			'is_callable'        => '', // If set, this callable will be be checked for availability to determine if a plugin is active.
		),
		array(
			'name'     				=> 'Slider Revolution', // The plugin name
			'slug'     				=> 'revslider', // The plugin slug (typically the folder name)
			'source'   				=> get_template_directory() . '/framework/plugins/revslider.zip', // The plugin source
			'required' 				=> true, // If false, the plugin is only 'recommended' instead of required
			'version' 				=> '5.4.5.1', // E.g. 1.0.0. If set, the active plugin must be this version or higher, otherwise a notice is presented
			'force_activation' 		=> false, // If true, plugin is activated upon theme activation and cannot be deactivated until theme switch
			'force_deactivation' 	=> false, // If true, plugin is deactivated upon theme switch, useful for theme-specific plugins
			'external_url' 			=> '', // If set, overrides default API URL and points to an external URL
			'is_callable'        => '', // If set, this callable will be be checked for availability to determine if a plugin is active.
		),
		array(
			'name' 		=> 'Contact Form 7',
			'slug' 		=> 'contact-form-7',
			'required' 	=> false,
		),

	);

	/*
	 * Array of configuration settings. Amend each line as needed.
	 *
	 * TGMPA will start providing localized text strings soon. If you already have translations of our standard
	 * strings available, please help us make TGMPA even better by giving us access to these translations or by
	 * sending in a pull-request with .po file(s) with the translations.
	 *
	 * Only uncomment the strings in the config array if you want to customize the strings.
	 */
	$config = array(
		'id'           => 'kreativa',                 // Unique ID for hashing notices for multiple instances of TGMPA.
		'default_path' => '',                      // Default absolute path to bundled plugins.
		'menu'         => 'tgmpa-install-plugins', // Menu slug.
		'has_notices'  => true,                    // Show admin notices or not.
		'dismissable'  => true,                    // If false, a user cannot dismiss the nag message.
		'dismiss_msg'  => '',                      // If 'dismissable' is false, this message will be output at top of nag.
		'is_automatic' => false,                   // Automatically activate plugins after installation or not.
		'message'      => '',                      // Message to output right before the plugins table.

		'strings'      => array(
			'page_title'                      => __( 'Install Required Plugins', 'kreativa' ),
			'menu_title'                      => __( 'Install Plugins', 'kreativa' ),
			/* translators: %s: plugin name. */
			'installing'                      => __( 'Installing Plugin: %s', 'kreativa' ),
			/* translators: %s: plugin name. */
			'updating'                        => __( 'Updating Plugin: %s', 'kreativa' ),
			'oops'                            => __( 'Something went wrong with the plugin API.', 'kreativa' ),
			'notice_can_install_required'     => _n_noop(
				/* translators: 1: plugin name(s). */
				'This theme requires the following plugin: %1$s.',
				'This theme requires the following plugins: %1$s.',
				'kreativa'
			),
			'notice_can_install_recommended'  => _n_noop(
				/* translators: 1: plugin name(s). */
				'This theme recommends the following plugin: %1$s.',
				'This theme recommends the following plugins: %1$s.',
				'kreativa'
			),
			'notice_ask_to_update'            => _n_noop(
				/* translators: 1: plugin name(s). */
				'The following plugin needs to be updated to its latest version to ensure maximum compatibility with this theme: %1$s.',
				'The following plugins need to be updated to their latest version to ensure maximum compatibility with this theme: %1$s.',
				'kreativa'
			),
			'notice_ask_to_update_maybe'      => _n_noop(
				/* translators: 1: plugin name(s). */
				'There is an update available for: %1$s.',
				'There are updates available for the following plugins: %1$s.',
				'kreativa'
			),
			'notice_can_activate_required'    => _n_noop(
				/* translators: 1: plugin name(s). */
				'The following required plugin is currently inactive: %1$s.',
				'The following required plugins are currently inactive: %1$s.',
				'kreativa'
			),
			'notice_can_activate_recommended' => _n_noop(
				/* translators: 1: plugin name(s). */
				'The following recommended plugin is currently inactive: %1$s.',
				'The following recommended plugins are currently inactive: %1$s.',
				'kreativa'
			),
			'install_link'                    => _n_noop(
				'Begin installing plugin',
				'Begin installing plugins',
				'kreativa'
			),
			'update_link' 					  => _n_noop(
				'Begin updating plugin',
				'Begin updating plugins',
				'kreativa'
			),
			'activate_link'                   => _n_noop(
				'Begin activating plugin',
				'Begin activating plugins',
				'kreativa'
			),
			'return'                          => __( 'Return to Required Plugins Installer', 'kreativa' ),
			'plugin_activated'                => __( 'Plugin activated successfully.', 'kreativa' ),
			'activated_successfully'          => __( 'The following plugin was activated successfully:', 'kreativa' ),
			/* translators: 1: plugin name. */
			'plugin_already_active'           => __( 'No action taken. Plugin %1$s was already active.', 'kreativa' ),
			/* translators: 1: plugin name. */
			'plugin_needs_higher_version'     => __( 'Plugin not activated. A higher version of %s is needed for this theme. Please update the plugin.', 'kreativa' ),
			/* translators: 1: dashboard link. */
			'complete'                        => __( 'All plugins installed and activated successfully. %1$s', 'kreativa' ),
			'dismiss'                         => __( 'Dismiss this notice', 'kreativa' ),
			'notice_cannot_install_activate'  => __( 'There are one or more required or recommended plugins to install, update or activate.', 'kreativa' ),
			'contact_admin'                   => __( 'Please contact the administrator of this site for help.', 'kreativa' ),

			'nag_type'                        => '', // Determines admin notice type - can only be one of the typical WP notice classes, such as 'updated', 'update-nag', 'notice-warning', 'notice-info' or 'error'. Some of which may not work as expected in older WP versions.
		),
	);

	tgmpa( $plugins, $config );
}